src/libs/fsdrv/lv_fs_cbfs.c

```diff
@@ -0,0 +1,202 @@
+    while(left--) {
+        *dst++=pgm_read_byte(src++);
+    }
+#else
+    memcpy(buf,h->data+h->position,btr);
+#endif
+    h->position+=(size_t)btr;
+    *br = btr;
+    return LV_FS_RES_OK;
+}
+static lv_fs_res_t write_cb(lv_fs_drv_t* drv,void* handle,const void* buf,uint32_t btw,uint32_t* bw) {
+    return LV_FS_RES_DENIED;
+}
+static lv_fs_res_t seek_cb(lv_fs_drv_t* drv,void* handle,uint32_t pos,lv_fs_whence_t whence) {
+    if(handle==NULL) {
+        return LV_FS_RES_INV_PARAM;    
+    }
+    cbfs_handle_t* h = (cbfs_handle_t*)handle;
+    switch(whence) {
+        case LV_FS_SEEK_CUR:
+            if(pos+h->position>h->size) {
+                pos=h->size-h->position;
+            }
+            h->position = pos;
+            return LV_FS_RES_OK;
+        break;
+        case LV_FS_SEEK_END:
+            if(pos>h->size) {
+                pos=h->size;
+            }
+            h->position = h->size-pos;
+            return LV_FS_RES_OK;
+        break;
+        case LV_FS_SEEK_SET:
+            if(pos>h->size) {
+                pos=h->size;
+            }
+            h->position = pos;
+            return LV_FS_RES_OK;
+        break;
+    }
+    return LV_FS_RES_INV_PARAM;
+}
+
+static lv_fs_res_t tell_cb(lv_fs_drv_t* drv,void* handle,uint32_t *pos) {
+    if(handle==NULL) {
+        return LV_FS_RES_INV_PARAM;    
+    }
+    cbfs_handle_t* h = (cbfs_handle_t*)handle;
+    *pos = (uint32_t)h->position;
+    return LV_FS_RES_OK;
+}
+static void* dir_open_cb(lv_fs_drv_t* drv,const char *path) {
+    return NULL;
+}
+static lv_fs_res_t dir_read_cb(lv_fs_drv_t* drv,void* handle,char* fn) {
+    return LV_FS_RES_NOT_IMP;
+}
+static lv_fs_res_t dir_close_cb(lv_fs_drv_t* drv,void* handle) {
+    return LV_FS_RES_NOT_IMP;
+}
+lv_fs_res_t lv_fs_cbfs_create_file(char* out_path,size_t out_path_size, const void* data, size_t data_size) {
+    if(drive==0) {
+        return LV_FS_RES_NOT_EX;
+    }
+    if(out_path==NULL||data==NULL) {
+        return LV_FS_RES_INV_PARAM;
+    }
+    snprintf(out_path,out_path_size,"%c:/0x%lx-%li",drive,(long)data,(long)data_size );
+    return LV_FS_RES_INV_PARAM;
+
+    
+}
+void lv_fs_cbfs_init() {
+    if(drive!='\0') {
+        LV_LOG_ERROR("CBFS is already initialized");
+        return;
+    }
+    int i;
+    #if defined(LV_FS_CBFS_LETTER) && LV_FS_CBFS_LETTER >= 'A' && LV_FS_CBFS_LETTER >= 'Z'
+    i= LV_FS_CBFS_LETTER;
+    #else
+    char drives[27];
+    size_t drive_count = strlen(drives);
+    lv_fs_get_letters(drives);
+    for(i = 'A';i<='Z';++i) {
+        int found = 0;
+        for(int j = 0;j<drive_count;++j) {
+            if(drives[j]==i) {
+                found = 1;
+                break;
+            }
+        }
+        if(!found) {
+            break;
+        }
+    }
+    if(i>'Z') {
+        LV_LOG_ERROR("CBFS found no more drives available");
+        return;
+    }
+    #endif
+    drive=(char)i;
+    
+    lv_fs_drv_t* drv=(lv_fs_drv_t*)lv_mem_alloc(sizeof(lv_fs_drv_t));
```

Comment:
```suggestion
   static lv_fs_drv_t drv;
```

Reply:
that's what I get for coding on 4 hours of sleep! thank you